partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeEnv.kt

```diff
@@ -57,7 +67,7 @@ internal class TypeEnv(public val schema: List<Rel.Binding>) {
      * @param name
      * @return
      */
-    private fun matchRoot(name: BindingName): Rex? {
+    private fun matchRoot(name: BindingName, level: Int): Rex? {
```

Comment:
Let's keep the naming consistent with the plan node, why not call this local frameRef you know?

I suggest "depth" in favor of frameRef, height, level, etc.